bgpd/bgp_vty.c

```diff
@@ -3019,6 +3019,74 @@ DEFUN (no_bgp_deterministic_med,
 	return CMD_SUCCESS;
 }
 
+static int bgp_inst_gr_config_vty(struct vty *vty, struct bgp *bgp, bool on)
+{
+	int ret = BGP_GR_FAILURE;
+
+	/*
+	 * Update the instance and all its peers, if appropriate.
+	 * Then, inform zebra of BGP's GR capabilities, if needed.
+	 */
+	ret = bgp_gr_update_all(bgp, on ? GLOBAL_GR_CMD : NO_GLOBAL_GR_CMD);
+	VTY_BGP_GR_ROUTER_DETECT_AND_SEND_CAPABILITY_TO_ZEBRA(bgp, bgp->peer,
+							      ret);
+	return ret;
+}
+
+static int bgp_global_gr_config_vty(struct vty *vty, bool on)
+{
+	struct listnode *node, *nnode;
+	struct bgp *bgp;
+	bool vrf_cfg = false;
+	int ret = BGP_GR_FAILURE;
+
+	if ((on && CHECK_FLAG(bm->flags, BM_FLAG_GR_RESTARTER)) ||
+	    (!on && !CHECK_FLAG(bm->flags, BM_FLAG_GR_RESTARTER)))
+		return CMD_SUCCESS;
+
+	/* See if GR is set per-vrf and warn user to delete */
+	if (!CHECK_FLAG(bm->flags, BM_FLAG_GR_CONFIGURED)) {
+		for (ALL_LIST_ELEMENTS(bm->bgp, node, nnode, bgp)) {
+			enum global_mode gr_mode = bgp_global_gr_mode_get(bgp);
+			if (gr_mode != GLOBAL_HELPER) {
```

Comment:
New line is missing after variable declaration.

Reply:
Fixed it